### ISSUE TITLE
How does ISDA work？

### ISSUE THREAD

**Issue by hheavenknowss** [blackfeather-wang/ISDA-for-Deep-Networks#5]
Hi，I'm reading your parper and your code, very useful work. but I'm wondering how does ISDA work exactly, I'm not fully understand it yet. My question is, does ISDA work while training a model? like training a detection model, and  mean while we use ISDA? or we use ISDA firstly, generate data, then we training the model?
I now thinking it is the first way to use, but I'm not sure. Am I right?
thanks a lot

**Comment by blackfeather-wang** [comment]
Thank you for your attention to our work. The final form of ISDA boils down to a new loss function, and it does not need to generate the augmented samples explicitly (although they can be visualized, as we show in the paper).

**Comment by hheavenknowss** [comment]
> Thank you for your attention to our work. The final form of ISDA boils down to a new loss function, and it does not need to generate the augmented samples explicitly (although they can be visualized, as we show in the paper).

Wow~that very cool, I don't even think about it can be a loss function, very impressive, I'm going to read this code againg

**Comment by blackfeather-wang** [comment]
Thank you again for your attention! We have submitted the camera-ready journal version of ISDA to Arxiv, where we added more convincing visualization results on ImageNet. You may see it tomorrow if everything goes well. :)
